extensions/etcdsnapshot/etcdsnapshot.go

```diff
@@ -203,34 +203,59 @@ func CreateRKE2K3SSnapshot(client *rancher.Client, clusterName string) error {
 }
 
 // RestoreRKE1Snapshot is a helper function to restore a snapshot on an RKE1 cluster. Returns error if any.
-func RestoreRKE1Snapshot(client *rancher.Client, clusterName string, snapshotRestore *management.RestoreFromEtcdBackupInput) error {
+func RestoreRKE1Snapshot(client *rancher.Client, clusterName string, snapshotRestore *management.RestoreFromEtcdBackupInput, initialControlPlaneValue, initialWorkerValue string) error {
 	clusterID, err := clusters.GetClusterIDByName(client, clusterName)
 	if err != nil {
 		return err
 	}
 
-	clusterResp, err := client.Management.Cluster.ByID(clusterID)
+	cluster, err := client.Management.Cluster.ByID(clusterID)
+	if err != nil {
+		return err
+	}
+
+	updatedCluster := cluster
+
+	updatedCluster.RancherKubernetesEngineConfig.UpgradeStrategy.MaxUnavailableControlplane = initialControlPlaneValue
+	updatedCluster.RancherKubernetesEngineConfig.UpgradeStrategy.MaxUnavailableWorker = initialWorkerValue
+
+	_, err = client.Management.Cluster.Update(cluster, updatedCluster)
 	if err != nil {
 		return err
 	}
 
 	logrus.Infof("Restoring snapshot: %v", snapshotRestore.EtcdBackupID)
-	err = client.Management.Cluster.ActionRestoreFromEtcdBackup(clusterResp, snapshotRestore)
+	err = client.Management.Cluster.ActionRestoreFromEtcdBackup(cluster, snapshotRestore)
 	if err != nil {
 		return err
 	}
 
+	err = wait.Poll(1*time.Second, defaults.FiveMinuteTimeout, func() (bool, error) {
```

Comment:
If I recall correctly wait.Poll is being deprecated. If possible could we update this?

Reply:
@igomez06 @slickwarren locally I tried this, but have to bump k8s.io/apiserver to 0.29.1. That introduces a number of changes to the `go.mod` and `go.sum` files. Not sure what practices for shepherd we're practicing with that.

Reply:
Sam is right, `PollWithContextTimeout` is causing a version bump?

Reply:
Resolved, I was testing on v2.7 where I saw this issue. We will fix that at a later time in that branch, but this branch has been updated.